Activities/Python/UiPath.Python.Activities/PythonScope.cs

```diff
@@ -109,91 +109,80 @@ protected override void CacheMetadata(NativeActivityMetadata metadata)
+                    if (autodetected != Version.Auto && autodetected != Version)
+                        throw new InvalidOperationException(string.Format(Resources.InvalidVersionException, Version.ToFriendlyString(), autodetected.ToFriendlyString()));
                 }
-            }
 
-            var workingFolder = WorkingFolder.Get(context);
-            if (!workingFolder.IsNullOrEmpty())
-            {
-                var dir = new DirectoryInfo(workingFolder);
-                if (!dir.Exists)
+                var workingFolder = WorkingFolder.Get(context);
+                if (!workingFolder.IsNullOrEmpty())
                 {
-                    var ex = new DirectoryNotFoundException(Resources.WorkingFolderPathInvalid);
-                    telemetryOperation?.SendWithException(ex);
-                    throw ex;
+                    var dir = new DirectoryInfo(workingFolder);
+                    if (!dir.Exists)
+                        throw new DirectoryNotFoundException(Resources.WorkingFolderPathInvalid);
+
+                    workingFolder = dir.FullName; //we need to pass an absolute path to the python host
                 }
-                workingFolder = dir.FullName; //we need to pass an absolute path to the python host
-            }
 
-            var operationTimeout = OperationTimeout.Get(context);
-            if (operationTimeout == 0)
-            {
-                operationTimeout = 3600; //default to 1h for no values provided.
-            }
+                var operationTimeout = OperationTimeout.Get(context);
+                if (operationTimeout == 0)
+                {
+                    operationTimeout = 3600; //default to 1h for no values provided.
+                }
 
-            try
-            {
-                await _pythonEngine.Initialize(workingFolder, cancellationToken, operationTimeout);
-            }
-            catch (Exception e)
-            {
-                Trace.TraceError($"Error initializing Python engine: {e}");
-                Exception cleanupEx = null;
                 try
                 {
-                    Cleanup();
+                    await _pythonEngine.Initialize(workingFolder, cancellationToken, operationTimeout);
                 }
-                catch (Exception cEx)
+                catch (Exception e)
                 {
-                    cleanupEx = cEx;
+                    Trace.TraceError($"Error initializing Python engine: {e}");
+                    Exception cleanupEx = null;
+                    try
+                    {
+                        Cleanup();
+                    }
+                    catch (Exception cEx)
+                    {
+                        cleanupEx = cEx;
+                    }
+
+                    var innerEx = cleanupEx != null ? new AggregateException(e, cleanupEx) : e;
+                    throw new InvalidOperationException(Resources.PythonInitializeException, innerEx);
```

Comment:
this catch clause seems a bit overcomplicated
```suggestion
                    
                catch (Exception e)
                {
                    Trace.TraceError($"Error initializing Python engine: {e}");
                    try
                    {
                        Cleanup();
                    }
                    catch (Exception cEx)
                    {
                        e = new AggregateException(e, cEx);
                    }

                    throw new InvalidOperationException(Resources.PythonInitializeException, e);
                }
```